File: src/main/java/com/bdev/hogwarts_api/rest_service/public_events/PublicEventsRestService.java
package com.bdev.hogwarts_api.rest_service.public_events;

import com.bdev.hogwarts_api.data.dto.events.Event;
import com.bdev.hogwarts_api.data.dto.events.EventParticipant;
import com.bdev.hogwarts_api.data.dto.events.EventType;

public interface PublicEventsRestService {
    Event getCurrentEvent(EventType eventType);
    void addEventParticipant(EventParticipant eventParticipant);
}

File: src/main/java/com/bdev/hogwarts_api/rest_service/public_events/PublicEventsRestService.kt
package com.bdev.hogwarts_api.rest_service.public_events

import com.bdev.hogwarts_api.data.dto.events.Event
import com.bdev.hogwarts_api.data.dto.events.EventParticipant
import com.bdev.hogwarts_api.data.dto.events.EventType

interface PublicEventsRestService {
    fun getCurrentEvent(eventType: EventType): Event
    fun addEventParticipant(eventParticipant: EventParticipant)
}
